napari/components/layerlist.py

```diff
@@ -98,7 +114,9 @@ def move_selected(self, index, insert):
         for insert_idx, elem_idx in enumerate(selected, start=insert - offset):
             indices.insert(insert_idx, elem_idx)
         # reorder list
-        self[:] = self[tuple(indices)]
+        with self.events.blocker():
```

Comment:
It looks like the events is `qt_layerlist.py` are still being triggered despite this block https://github.com/napari/napari/pull/1504/files#diff-6408ff7cd9e06a487f66df8d17583e6eR64-R66 not sure why though

Reply:
The blocker does seem to stop event listeners added after the layerlist is created, say in the console, so not quite sure what's happening here

Reply:
i'll take a look... can you point me to code I can use to reproduce seeing the `qt_layerlist` emit events despite the block?

Reply:
Not at my keyboard right now, but I added print statements inside the listeners in that file (the two where layer widgets get removed / added respectively) then loaded a napari viewer with a bunch of layers, did a manual drag and drop and saw those print statements appear inside the blocker context - which to me was unexpected

Reply:
print statements were https://github.com/napari/napari/blob/master/napari/_qt/qt_layerlist.py#L95 and here https://github.com/napari/napari/blob/master/napari/_qt/qt_layerlist.py#L79, but i also see it with print statements here https://github.com/napari/napari/blob/master/napari/_qt/qt_controls.py#L66 and https://github.com/napari/napari/blob/master/napari/_qt/qt_controls.py#L79.

If I do `viewer.layers.events.connect(lambda e: print(f"you called {e.type}"))` then this print statement is blocked by the blocker.

I wonder if the blocker blocks `viewer.layers.events.connect` but doesn't block individual ``viewer.layers.events.insertedconnect` calls. I will test now!!

Reply:
Yes, this appears to be the case `with self.events.blocker():` will block

```python
viewer.layers.events.connect(lambda e: print(f"you called {e.type}"))
```

but not block

```python
viewer.layers.events.inserted.connect(lambda e: print(f"you called ONLY INSERTED"))
```

I think this is a bug and it should block that as well. I think with that fixed this PR would then work as expected.

Thoughts @tlambert03 - can you reproduce?

Reply:
I can reproduce this.  let me figure this one out before we try to work around it